### DISCUSSION TITLE
[OP Stack Deployment] Deploy Failed due to Exceeds Tx Fee

### DISCUSSION THREAD

**Discussion by yusufsuhair** [ethereum-optimism/developers#410] [Deploying the OP Stack]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

Hi all,

I tried to deploy the L1 contracts using this command forge script `scripts/Deploy.s.sol:Deploy --private-key $GS_ADMIN_PRIVATE_KEY --broadcast --rpc-url $L1_RPC_URL --slow`

But then, after transaction number 20, it failed due to the next deployment cost me 1.10 ETH

`Sending transactions [20 - 20].`

`Transactions saved to: /root/optimism/packages/contracts-bedrock/broadcast/Deploy.s.sol/11155111/run-latest.json`

`Sensitive values saved to: /root/optimism/packages/contracts-bedrock/cache/Deploy.s.sol/11155111/run-latest.json`

`Error: Failed to send transaction`

`Context: `
`- (code: -32000, message: tx fee (1.10 ether) exceeds the configured cap (1.00 ether), data: None)`

so I tried again to redeploy, but it cannot continue from the previous giving this error instead:

`== Logs == 
Connected to network with chainid 11155111 Storing temp deployment data in /root/optimism/packages/contracts-bedrock/deployments/getting-started/.deploy DeployConfig: reading file /root/optimism/packages/contracts-bedrock/deploy-config/getting-started.json Deploying from Deploy Deployment context: getting-started Deploying a fresh OP Stack including SuperchainConfig Deploying Safe New SystemOwnerSafe deployed at 0xb39fdb3F3E9D10f76AeF55dedb25ca19cdD6e7d6 Setting up Superchain Deploying AddressManager AddressManager deployed at 0x1005Ed3B73834005fca937cfA4Fbb90CDa2B326d Deploying ProxyAdmin ProxyAdmin deployed at 0xfA66E3850fa96996d30648c86C1Ea047beeC9598 ProxyAdmin ownership transferred to Safe at: 0xb39fdb3F3E9D10f76AeF55dedb25ca19cdD6e7d6 Deploying ERC1967 proxy forSuperchainConfigProxy at 0x2cCb8342Fb4F444269D96aBef291Db3c7b83A0A9 Error: script failed: <empty revert data>`

How can I fix this, few months ago it was fine.

Should I change the RPC ?

Currency I'm using this config inside `.envrc`:

`export L1_RPC_KIND=any`
`export L1_RPC_URL=https://rpc.sepolia.org`

Please help.

**Comment by opfocus** [comment]
At present, the gas on Sepolia is very high, you should first estimate whether there is enough balance to deploy them: a simple way to estimate is to prepare a balance slightly higher than 40 million * gasPrice. 
If you want to continue with the already deployed contract, you need to modify the deployment file deploy.s.sol

`code: -32000, message: tx fee (1.10 ether) exceeds the configured cap (1.00 ether), data: None`   about this, I dont know the reason CC @Chomtana

**Comment by Chomtana** [comment]
It's a problem that the foundry doesn't cache deployment like the old hardhat implementation that you may have used a few months ago. To proceed with foundry deployment, you can do like @opfocus recommend:

1. Comment deployed part in
2. `IMPL_SALT` must be the same

Exceeds the configured cap error is from the RPC side. You can try changing to use other RPC provider such as Quicknode / Alchemy / etc (Don't sure which one allow unlimited gas cap)

https://ethereum.stackexchange.com/questions/99034/tx-fee-2-11-ether-exceeds-the-configured-cap-1-00-ether-how-to-bypass-defau
